pdns/tcpreceiver.cc

```diff
@@ -809,6 +809,12 @@ int TCPNameserver::doAXFR(const DNSName &target, shared_ptr<DNSPacket> q, int ou
     }
   }
 
+  // Group records by name and type, signpipe stumbles over interrupted rrsets
+  sort(zrrs.begin(), zrrs.end(), [](const DNSZoneRecord& a, const DNSZoneRecord& b) {
+    if (a.dr.d_name != b.dr.d_name) return a.dr.d_name < b.dr.d_name;
```

Comment:
I've been burned by messing this up, I suggest: return std::tie(a.dr.d_name, b.dr.d_name) < std::tie(a.dr.d_type, b.dr.d_type); I know you got the code right, but when I have to debug something I have to run the cleverness on my brain and the idiomatic std::tie construct I don't have to think about